File: 1392-find-the-difference-of-two-arrays/find-the-difference-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
    let n1set = new Set(nums1);
    let n2set = new Set(nums2);
    let n1 = [];
    let n2 = [];
    for (let n of n1set) {
        if (!n2set.has(n)) n1.push(n);
    }
    for (let n of n2set) {
        if (!n1set.has(n)) n2.push(n);
    }
    return [n1, n2];
};

File: 1392-find-the-difference-of-two-arrays/find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        nums1_vals = set(nums1)
        nums2_vals = set(nums2)

        for n1 in nums1:
            if n1 in nums2_vals:
                nums1_vals.remove(n1)
                nums2_vals.remove(n1)
        
        return [list(nums1_vals), list(nums2_vals)]
        